### PR TITLE
Improve error logging for  in Azure Pipelines Scaler

### LINKED ISSUE #3996: Azure Pipelines Scaler: Improve error logging using poolID
### Proposal

Currently, in case of any error validating the poolID, the error is hidden:
```go
body, err := getAzurePipelineRequest(ctx, url, metadata, httpClient)
	if err != nil {
		return -1, fmt.Errorf("agent pool with id `%s` not found", poolID)
	}

```
This makes impossible to troubleshoot the reason behind the fails, we should just return the original error

### Use-Case

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dttung2905**
Hi team,

This PR will improve the error logging in Azure Pipelines Scaler. Could you help to review it :pray: ?

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #3996

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ Here is an overview of all new **experimental** features:
 - **RabbitMQ Scaler**:  Add TLS support ([#967](https://github.com/kedacore/keda/issues/967))
 - **Redis Scalers**: Add support to Redis 7 ([#4052](https://github.com/kedacore/keda/issues/4052))
 - **Selenium Grid Scaler**: Add 'platformName' to selenium-grid scaler metadata structure ([#4038](https://github.com/kedacore/keda/issues/4038))
+- **Azure Pipelines Scaler**: Improve error logging for `validatePoolID` ([#3996](https://github.com/kedacore/keda/issues/3996))
```
Comment (JorTurFer) [inline CHANGELOG.md:69]: Could you move this below General? We put always General on top of any section, and below the other elements are sorted alphabetically. This line should be between General and Hashicopr

**Review by JorTurFer — APPROVED** [review]
LGTM!

File: CHANGELOG.md (hunk 1 above)
Comment (dttung2905) [reply CHANGELOG.md]: Thanks for the quick review and feedback. I have pushed this line above, right below `General` :pray:

**Review by zroubalik — APPROVED** [review]
LGTM

**Pull request merged**